common/queue_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewQueue(t *testing.T) {
+	q := NewQueue[int]()
+
+	assert.NotNil(t, q, "NewQueue() should return a non-nil queue.")
+	assert.True(t, q.IsEmpty(), "A new queue should be empty.")
+	assert.Equal(t, 0, q.Len(), "A new queue should have a size of 0.")
+}
+
+func TestQueue_Push(t *testing.T) {
+	q := NewQueue[int]()
+
+	q.Push(4)
+	q.Push(5)
+
+	assert.Equal(t, 4, q.Peek())
+	assert.False(t, q.IsEmpty())
+}
+
+func TestQueue_Pop(t *testing.T) {
```

Comment:
This test could be simplified - it has multiple "Act" blocks for instance. I'd recommend splitting it into two tests.